example/spec/serial/ingestPdrWithNodeNameFtpSpec.js

```diff
@@ -0,0 +1,422 @@
+  updateAndUploadTestDataToBucket,
+} = require('../helpers/testUtils');
+
+const {
+  loadFileWithUpdatedGranuleIdPathAndCollection,
+} = require('../helpers/granuleUtils');
+
+const { waitForApiStatus, waitForModelStatus } = require('../helpers/apiUtils');
+const { deleteProvidersByHost, waitForProviderRecordInOrNotInList } = require('../helpers/Providers');
+
+const lambdaStep = new LambdaStep();
+const workflowName = 'DiscoverAndQueuePdrs';
+const origPdrFilename = 'MOD09GQ_1granule_v3_with_NODE_NAME.PDR';
+
+const s3data = [
+  '@cumulus/test-data/pdrs/MOD09GQ_1granule_v3_with_NODE_NAME.PDR',
+];
+
+describe('Ingesting from PDR', () => {
+  const providersDir = './data/providers/s3/';
+  const collectionsDir = './data/collections/s3_MOD09GQ_006';
+
+  let addedCollection;
+  let beforeAllFailed;
+  let config;
+  let executionModel;
+  let nodeNameProviderId;
+  let parsePdrExecutionArn;
+  let pdrFilename;
+  let provider;
+  let testDataFolder;
+  let testDataGranuleId;
+  let testSuffix;
+  let workflowExecution;
+  let functionName;
+  let testFilePaths;
+
+  const ingestTime = Date.now() - 1000 * 30;
+
+  beforeAll(async () => {
+    try {
+      config = await loadConfig();
+      functionName = `${config.stackName}-populateTestLambda`;
+      process.env.ExecutionsTable = `${config.stackName}-ExecutionsTable`;
+      process.env.PdrsTable = `${config.stackName}-PdrsTable`;
+      executionModel = new Execution();
+
+      const testId = createTimestampedTestId(config.stackName, 'IngestFromPdrWithNodeName');
+      testSuffix = createTestSuffix(testId);
+      pdrFilename = `${testSuffix.slice(1)}_${origPdrFilename}`;
+      provider = { id: `s3_provider${testSuffix}` };
+      testDataFolder = createTestDataPath(testId);
+
+      const ftpProvider = await buildFtpProvider(`${randomString(4)}-${testSuffix}`);
+      await deleteProvidersByHost(config.stackName, config.pdrNodeNameProviderBucket);
+      await deleteProvidersByHost(config.stackName, ftpProvider.host);
+
+      nodeNameProviderId = `provider-${randomString(4)}-${testSuffix}`;
+
+      await providersApi.createProvider({
+        prefix: config.stackName,
+        provider: {
+          id: nodeNameProviderId,
+          protocol: 's3',
+          host: config.pdrNodeNameProviderBucket,
+        },
+      });
+
+      // Create FTP provider
+      await providersApi.createProvider({
+        prefix: config.stackName,
+        provider: ftpProvider,
+      });
+
+      await Promise.all([
+        waitForProviderRecordInOrNotInList(config.stackName, nodeNameProviderId, true, { timestamp__from: ingestTime }),
+        waitForProviderRecordInOrNotInList(config.stackName, ftpProvider.id, true, { timestamp__from: ingestTime }),
+      ]);
+
+      let testData;
+      try {
+        testData = await lambda().invoke({
+          FunctionName: functionName,
+          Payload: JSON.stringify({
+            prefix: config.stackName,
+          }),
+        }).promise();
```

Comment:
I'm being a bit lazy, but why does the test need to populate files on FTP in the setup? Why can't the files permanently exist on the FTP server? I guess because that's fundamentaly not possible?

And is the bug we're fixing fundamentally related to FTP servers?

Reply:
No, it's fundamentally related to providers with auth credentials, and it's perhaps lazy (efficient?) but ftp was the easiest server to update directly through it's data acquisition interface.

Reply:
Would it be equally feasible to add basic auth to our fake HTTP provider? Then we wouldn't have to run a Lambda to populate data on FTP.

Could we not add basic auth to Apache somewhere in here?

https://github.com/nasa/cumulus/blob/master/example/fake-provider-cf.yml#L107

Reply:
>Would it be equally feasible to add basic auth to our fake HTTP provider? Then we wouldn't have to run a Lambda to populate data on FTP.

I'm not sure I understand what you mean.  Even if we add auth to the HTTP provider,  it'd have to provide authenticated access outside the VPC to allow integration tests to update the data, and that's ... fraught?     

An approach that would get us out of the lambda invocation business might be mounting a s3 bucket to the https host, but that initially felt like it might be more work.    

> I'm being a bit lazy, but why does the test need to populate files on FTP in the setup? 

It's fair that it doesn't *have* to, but generally ingest tests that reuse granuleIds are something we've tried to avoid for legit collision reasons. 

>Why can't the files permanently exist on the FTP server? I guess because that's fundamentally not possible?

I.... don't think it's ever *not* possible to put data on a data volume/bucket?  This is entirely to setup dynamic granule generation in the most straightforward way possible.